src/org/wso2/tg/jenkins/util/ConfigUtils.groovy

```diff
@@ -181,3 +190,27 @@ def addToInputParams(yamlContent, key, value, section) {
     }
 }
 
+/**
+ * Resolves the values if they are sensitive parameters stored as Jenkins credentials
+ * Ex:
+ * infra.secretValue=CREDENTIALS(secret-key)
+ *
+ * if the Value is presented in the above manner, secret-key will be used to retrieve the
+ * parameter from Credentials plugin
+ *
+ * @param value user defined value
+ * @return resolved parameter
+ */
+def resolveCredentials(value){
+    def common = new Common()
+    if (value.startsWith("CREDENTIALS(") && value.endsWith(")")){
+        def split = value.split("[()]")
```

Comment:
Can there be a chance to have `( )` inside a secret-key?

Reply:
ideally `( ) `should not be used within a key. It's possible its present in the value.
in here we only provide the key value to retrieve the secret value  from credentials plugin.
Here TestGrid team has the control over the key names as the credentials are added via an admin.